openapi/spec.go

```diff
@@ -9,16 +9,18 @@ type OpenAPI struct {
 	Paths      Paths       `json:"paths" yaml:"paths"`
 	Components *Components `json:"components,omitempty" yaml:"components,omitempty"`
 	Tags       []*Tag      `json:"tags,omitempty" yaml:"tags,omitempty"`
+	Security   []*Security `json:"security,omitempty" yaml:"security,omitempty"`
 }
 
 // Components holds a set of reusable objects for different
 // ascpects of the specification.
 type Components struct {
-	Schemas    map[string]*SchemaOrRef    `json:"schemas,omitempty" yaml:"schemas,omitempty"`
-	Responses  map[string]*ResponseOrRef  `json:"responses,omitempty" yaml:"responses,omitempty"`
-	Parameters map[string]*ParameterOrRef `json:"parameters,omitempty" yaml:"parameters,omitempty"`
-	Examples   map[string]*ExampleOrRef   `json:"examples,omitempty" yaml:"examples,omitempty"`
-	Headers    map[string]*HeaderOrRef    `json:"headers,omitempty" yaml:"headers,omitempty"`
+	Schemas         map[string]*SchemaOrRef     `json:"schemas,omitempty" yaml:"schemas,omitempty"`
+	Responses       map[string]*ResponseOrRef   `json:"responses,omitempty" yaml:"responses,omitempty"`
+	Parameters      map[string]*ParameterOrRef  `json:"parameters,omitempty" yaml:"parameters,omitempty"`
+	Examples        map[string]*ExampleOrRef    `json:"examples,omitempty" yaml:"examples,omitempty"`
+	Headers         map[string]*HeaderOrRef     `json:"headers,omitempty" yaml:"headers,omitempty"`
+	SecuritySchemes map[string]*SecuritySchemes `json:"securitySchemes,omitempty" yaml:"securitySchemes,omitempty"`
```

Comment:
According to the OpenAPI spec, the `securitySchemes` fields of the `Components` types can be a Security Scheme Object or a Reference Object:
```
Map[string, Security Scheme Object | Reference Object]
```